diff --git a/ibex_core.core b/ibex_core.core
index fb9eca5..bda0468 100644
--- a/ibex_core.core
+++ b/ibex_core.core
@@ -31,8 +31,8 @@ filesets:
       # XXX: Figure out the best way to switch these two implementations
       # dynamically on the target.
 #      - rtl/ibex_register_file_latch.sv # ASIC
-#      - rtl/ibex_register_file_fpga.sv # FPGA
-      - rtl/ibex_register_file_ff.sv # generic FF-based
+      - rtl/ibex_register_file_fpga.sv # FPGA
+#      - rtl/ibex_register_file_ff.sv # generic FF-based
       - rtl/ibex_core.sv
     file_type: systemVerilogSource
 
diff --git a/rtl/ibex_core.sv b/rtl/ibex_core.sv
index 707eb5c..173e4dd 100644
--- a/rtl/ibex_core.sv
+++ b/rtl/ibex_core.sv
@@ -15,7 +15,7 @@
 module ibex_core #(
     parameter bit          PMPEnable                = 1'b0,
     parameter int unsigned PMPGranularity           = 0,
-    parameter int unsigned PMPNumRegions            = 4,
+    parameter int unsigned PMPNumRegions            = 1,
     parameter int unsigned MHPMCounterNum           = 0,
     parameter int unsigned MHPMCounterWidth         = 40,
     parameter bit          RV32E                    = 1'b0,
diff --git a/shared/rtl/ram_1p.sv b/shared/rtl/ram_1p.sv
index 244c390..1c8b581 100644
--- a/shared/rtl/ram_1p.sv
+++ b/shared/rtl/ram_1p.sv
@@ -83,5 +83,102 @@ module ram_1p #(
       $display("Initializing SRAM from %s", MEM_FILE);
       $readmemh(MEM_FILE, mem);
     end
+  `else
+    initial begin
+      // ////////////////////////////////////////////////////
+      //mem[0] = 32'h0000C7B7;
+      //mem[1] = 32'h00A00713; // leds: 0xa: on off on off
+      //mem[1] = 32'h00900713; // leds: 0x9: on off off on
+      //mem[2] = 32'h00E78823;
+      //mem[3] = 32'h0000006F;
+      // ////////////////////////////////////////////////////
+      mem[ 0] = 32'h00000013;
+      mem[ 1] = 32'h00000013;
+      mem[ 2] = 32'h00000013;
+      mem[ 3] = 32'h00000013;
+      mem[ 4] = 32'h00000013;
+      mem[ 5] = 32'h00000013;
+      mem[ 6] = 32'h00000013;
+      mem[ 7] = 32'h00000013;
+      mem[ 8] = 32'h00000013;
+      mem[ 9] = 32'h00000013;
+
+      mem[10] = 32'h00000013;
+      mem[11] = 32'h00000013;
+      mem[12] = 32'h00000013;
+      mem[13] = 32'h00000013;
+      mem[14] = 32'h00000013;
+      mem[15] = 32'h00000013;
+      mem[16] = 32'h00000013;
+      mem[17] = 32'h00000013;
+      mem[18] = 32'h00000013;
+      mem[19] = 32'h00000013;
+
+      mem[20] = 32'h00000013;
+      mem[21] = 32'h00000013;
+      mem[22] = 32'h00000013;
+      mem[23] = 32'h00000013;
+      mem[24] = 32'h00000013;
+      mem[25] = 32'h00000013;
+      mem[26] = 32'h00000013;
+      mem[27] = 32'h00000013;
+      mem[28] = 32'h00000013;
+      mem[29] = 32'h00000013;
+
+      mem[30] = 32'h00000013;
+      mem[31] = 32'h0100006F;
+      mem[32] = 32'h0100006F;
+      mem[33] = 32'h0080006F;
+      mem[34] = 32'h0040006F;
+      mem[35] = 32'h0000006F;
+      mem[36] = 32'h00000093;
+      mem[37] = 32'h81868106;
+      mem[38] = 32'h82868206;
+      mem[39] = 32'h83868306;
+
+      mem[40] = 32'h84868406;
+      mem[41] = 32'h85868506;
+      mem[42] = 32'h86868606;
+      mem[43] = 32'h87868706;
+      mem[44] = 32'h88868806;
+      mem[45] = 32'h89868906;
+      mem[46] = 32'h8A868A06;
+      mem[47] = 32'h8B868B06;
+      mem[48] = 32'h8C868C06;
+      mem[49] = 32'h8D868D06;
+
+      mem[50] = 32'h8E868E06;
+      mem[51] = 32'h8F868F06;
+      mem[52] = 32'h00010117;
+      mem[53] = 32'hF3010113;
+      mem[54] = 32'h13C00D13;
+      mem[55] = 32'h13C00D93;
+      mem[56] = 32'h01BD5763;
+      mem[57] = 32'h000D2023;
+      mem[58] = 32'hDDE30D11;
+      mem[59] = 32'h0513FFAD;
+
+      mem[60] = 32'h05930000;
+      mem[61] = 32'h00EF0000;
+      mem[62] = 32'h47290040;
+      mem[63] = 32'h882367B1;
+      mem[64] = 32'h673700E7;
+      mem[65] = 32'h0713005F;
+      mem[66] = 32'h66B1E107;
+      mem[67] = 32'h000187BA;
+      mem[68] = 32'h00010001;
+      mem[69] = 32'h17FD0001;
+
+      mem[70] = 32'hC783FBFD;
+      mem[71] = 32'hC7930106;
+      mem[72] = 32'hF793FFF7;
+      mem[73] = 32'h88230FF7;
+      mem[74] = 32'hB7CD00F6;
+      mem[75] = 32'h00000000;
+      mem[76] = 32'h00000000;
+      mem[77] = 32'h00000000;
+      mem[78] = 32'h00000000;
+      // ////////////////////////////////////////////////////
+    end
   `endif
 endmodule
